### PR TITLE
Plugin support (part 1)

### PR DESCRIPTION
**Pull request by aarzilli**
```
proc,proc/native,proc/gdbserial: initial plugin support

Adds initial support for plugins, this is only the code needed to keep
track of loaded plugins on linux (both native and gdbserial backend).

It does not actually implement support for debugging plugins on linux.

Updates #865

```

### CODE REVIEW

**Comment by webern** [comment]
Thank you for working on support of Go plugins. I need to use that feature in my project and I'm hoping that Delve can support them sometime soon.

**Comment by aarzilli** [comment]
Rebased

File: pkg/proc/linutil/dynamic.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+	"github.com/go-delve/delve/pkg/proc"
+)
+
+const (
+	maxNumLibraries      = 1000000 // maximum number of loaded libraries, to avoid loading forever on corrupted memory
+	maxLibraryPathLength = 1000000 // maximum length for the path of a library, to avoid loading forever on corrupted memory
+)
+
+var ErrTooManyLibraries = errors.New("too many libraries")
```
Comment (derekparker) [inline pkg/proc/linutil/dynamic.go]: nit: I think we can make this error a little more descriptive, e.g. "maximum number of shared libraries loaded" or something.

File: pkg/proc/linutil/dynamic.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+var ErrStringTooLong = errors.New("string too long")
```
Comment (derekparker) [inline pkg/proc/linutil/dynamic.go]: I think we can also make this more descriptive, unless there is context coming from a wrapping error.

File: pkg/proc/linutil/dynamic.go [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+			break
+		}
+		r = append(r, buf[0])
+		addr++
+	}
+	return string(r), nil
+}
+
+// ElfUpdateSharedObjects reads the list of dynamic libraries loaded by the
+// dynamic linker from the .dynamic section and uses it to updated p.BinInfo().
```
Comment (derekparker) [inline pkg/proc/linutil/dynamic.go]: s/updated/update/

**Pull request merged**